src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1358,11 +1354,14 @@ private void CheckOpen()
             }
         }
 
-        private void CheckExistenceOfDefaultCollection()
+        private void CheckDefaultCollectionOpen()
         {
-            if (DefaultCollection == null) {
-                throw new InvalidOperationException($"Default Collection is deleted from the database { ToString() }.");
-            }
+            ThreadSafety.DoLocked(() =>
+            {
+                if (DefaultCollection.IsClosed) {
```

Comment:
Not good enough.  DefaultCollection can be null and now you've thrown a null reference exception.  Why not just have your internal property (`DefaultCollection`) throw an exception if it is null or closed, rather than calling CheckDefaultCollectionOpen everywhere.